### PR TITLE
feat: assigning schema id to parameter schemas in components

### PR DESCRIPTION
**Pull request by jonaslagoni**
**Description**
This PR fixes an issue where we assign parameters wrong ids.

**Related issue(s)**
Fixes https://github.com/asyncapi/parser-js/issues/325

### CODE REVIEW

**Comment by KhudaDad414** [comment]
Well, that was fast. :)

File: lib/anonymousNaming.js [hunk 1]
```diff
@@ -45,6 +47,22 @@ function assignUidToComponentSchemas(doc) {
...
+ * Assign uid to component parameters schemas
+ * 
+ * @private
+ * @param {AsyncAPIDocument} doc 
+ */
+function assignUidToComponentParameterSchemas(doc) {
+  if (doc.hasComponents()) {
+    for (const [parameterKey, parameter] of Object.entries(doc.components().parameters())) {
+      if (parameter.schema()) {
+        parameter.schema().json()[String(xParserSchemaId)] = parameterKey;
```
Comment (derberg) [inline lib/anonymousNaming.js:61]: feels like this entire loop is the same as in line 29, just the place where param is taken from is different. Can we do something about it?

**Review by derberg — CHANGES_REQUESTED** [review]
Probably title should be something like `feat: assigning schema id to parameter schemas in components` and the other fix could be in description

File: lib/anonymousNaming.js (hunk 1 above)
Comment (jonaslagoni) [reply lib/anonymousNaming.js]: Added reusability 😄

File: lib/anonymousNaming.js [hunk 2]
```diff
@@ -17,6 +17,19 @@ function assignNameToComponentMessages(doc) {
   }
 }
 
+/**
+ * Assign ids based on parameter keys.
+ * 
+ * @param {Record<string,Schema>} parameterObject
```
Comment (derberg) [inline lib/anonymousNaming.js:23]:
```suggestion
 * @private
 * @param {Record<string,Schema>} parameterObject 
```

**Review by derberg — APPROVED** [review]

**Pull request merged**